GPyOpt/core/task/objective.py

```diff
@@ -82,22 +71,23 @@ def _syncronous_batch_evaluation(self,x):
         Evaluates the function a x, where x can be a single location or a batch. The evaluation is performed in parallel
         according to the number of accessible cores.
         """
-        from multiprocessing import Process, Pipe
-
+        from sys import version_info
         # --- parallel evaluation of the function
         divided_samples = [x[i::self.n_procs] for i in range(self.n_procs)]
-        pipe = [Pipe() for i in range(self.n_procs)]
-        proc = [Process(target=spawn(self._eval_func),args=(c,k)) for k,(p,c) in zip(divided_samples,pipe)]
-        [p.start() for p in proc]
-        [p.join() for p in proc]
-
-        # --- time of evaluation is set to constant (=1). This is one of the hypothesis of synchronous batch methods.
-        f_evals = np.zeros((x.shape[0],1))
-        cost_evals = np.ones((x.shape[0],1))
-        i = 0
-        for (p,c) in pipe:
-            f_evals[i::self.n_procs] = p.recv()[0] # throw away costs
-            i += 1
+
+        if version_info.major < 3:
+            from multiprocess import Pool
+            p = Pool(processes=self.n_procs)
+            results = p.map(self._eval_func, divided_samples)
+            p.close()
+            p.join()
+        else:
+            from joblib import Parallel, delayed
+            results = Parallel(n_jobs=self.n_procs)([delayed(self._eval_func)(args) for args in divided_samples])
+
+        f_evals = np.vstack([res[0] for res in results])
+        cost_evals = np.hstack([res[1] for res in results])
```

Comment:
Same somment - np.array and transpose should do the job, if I am not mistaken. hstack and vstack are now considered deprecated in NumPy and are only supported for compatibility reasons.